azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -172,40 +172,50 @@ def create_from_edge_environment(cls):
             try:
                 connection_string = os.environ["EdgeHubConnectionString"]
                 ca_cert_filepath = os.environ["EdgeModuleCACertificateFile"]
-            except KeyError:
-                # TODO: consider using a different error here. (OSError?)
-                raise auth.IoTEdgeError("IoT Edge environment not configured correctly")
-
+            except KeyError as e:
+                new_err = OSError("IoT Edge environment not configured correctly")
+                new_err.__cause__ = e
```

Comment:
> new_err.__cause__ = e [](start = 15, length = 22)

I always get worried setting dunders.  Is this copasetic?

Reply:
Yes. See PEP3134 ([https://www.python.org/dev/peps/pep-3134/#id37](https://www.python.org/dev/peps/pep-3134/#id37))

---
In reply to: [326735919](https://github.com/Azure/azure-iot-sdk-python/pull/309#discussion_r326735919) [](ancestors = 326735919)